frontend/client-list/table/desktop-clients-table.component.tsx

```diff
@@ -181,6 +182,28 @@ export default function DesktopClientsTable(props: ClientsTableProps) {
       </span>
     );
   }
+
+  function handleCheckBoxChange(id, data) {
+    const newIsSelected = props.isSelected;
+    const newSelection = props.selectedClients;
+    if (props.isSelected[id] === true) {
+      delete newIsSelected[id];
+      props.setIsSelected(newIsSelected);
+      const objectPosition = newSelection
+        .map(client => {
+          console.log(client);
+          return client[id].id;
+        })
+        .indexOf(id);
+      newSelection.splice(objectPosition, 1);
+      props.setSelectedClients(newSelection);
+    } else {
+      newIsSelected[id] = true;
+      newSelection.push(data);
```

Comment:
It's best not to modify props by pushing data into the array. From the react docs:

> Remember that props are readonly. They should not be modified in any way:

https://reactjs.org/docs/glossary.html#props

Instead, you could do this:
```js
props.setSelectedClients(props.selectedClients.concat(data))
```

which does not modify the original array